yt/data_objects/tests/test_profiles.py

```diff
@@ -356,39 +356,39 @@ def tearDown(self):
     def test_unequal_data_shape_profile(self):
         density = np.random.random(128)
         temperature = np.random.random(128)
-        cell_mass = np.random.random((128, 128))
+        mass = np.random.random((128, 128))
 
         my_data = {
-            "density": density,
-            "temperature": temperature,
-            "cell_mass": cell_mass}
+            ('gas', 'density'): density,
```

Comment:
This isn't going to result in the field type being saved as `gas`. It will still be `data` unless you pass the `field_types` keyword the way I mentioned in the earlier comment.

Reply:
I know.  My last comment explained that I switched the behavior partially back on this.  The problem is that for some reason the existing tests need to match both `('all', 'mass')` as well as `('gas', 'mass')` in the HDF5 output.  It also made me realize that if I set the default weighting to `('gas', 'mass')`, `PhasePlot` doesn't really make sense for any non-gas field.  So I just changed it to `mass`, which not only fixes the tests but it also means that the weight field is ambiguous enough to go with the available `mass` field.

Reply:
Ok, but this code here doesn't do what it suggests so would be misleading to anyone reading it. The tuple as keys in `my_data` should be removed.

Reply:
Is this conversation resolved?